metomi/isodatetime/data.py

```diff
@@ -522,54 +542,50 @@ def __floordiv__(self, other):
                 "'%s' should be integer." %
                 type(other).__name__
             )
-        new = self.copy()
+        new = self._copy()
         if self.get_is_in_weeks():
-            new.weeks //= other
+            new._weeks //= other
             return new
-        new.years //= other
-        new.months //= other
-        new.days //= other
-        new.hours //= other
-        new.minutes //= other
-        new.seconds //= other
+        new._years //= other
+        new._months //= other
+        new._days //= other
+        new._hours //= other
+        new._minutes //= other
+        new._seconds //= other
         return new
 
+    def __hash__(self) -> int:
+        return hash(self.get_days_and_seconds())
+
     def __eq__(self, other: "Duration") -> bool:
-        my_data = self.get_days_and_seconds()
-        other_data = other.get_days_and_seconds()
-        return my_data == other_data
+        # TODO: check instance of other
+        return self.get_days_and_seconds() == other.get_days_and_seconds()
 
```

Comment:
A possible problem with this is that months & years are ambiguous lengths of time. Nevertheless, the existing implementation already meant that e.g. `P1M == P30D`.

Reply:
No problem, it's what the code did before and I think Python's `__eq__` is meant to be more like `=` than `≡` anyway.